daml/Marketplace/Distribution/Auction/Utils.daml

```diff
@@ -0,0 +1,77 @@
+module Marketplace.Distribution.Auction.Utils where
+
+import DA.Finance.Asset (AssetDeposit, AssetDeposit_Split(..))
+import Marketplace.Settlement qualified as Settlement
+import Marketplace.Distribution.Auction.Model qualified as Auction
+import Marketplace.Distribution.Bid.Model qualified as Bid
+import Marketplace.Distribution.Bid.Model (Bid(..))
+import Marketplace.Distribution.Bid.Service qualified as Bid
+import Marketplace.Trading.AllocationAccountRule qualified as Allocation
+import Marketplace.Trading.AllocationAccountRule (AllocationAccountRule(..))
+
+-- | Takes a list and splits it based off a predicate. Values which match the predicate go in the right list of the tuple
+splitList : (a -> Bool) -> [a] -> ([a], [a])
+splitList p xs = foldr (\x (y,z) -> if p x then (y, x :: z) else (x :: y, z)) ([],[]) xs
```

Comment:
This function could output `[Either a a]` then the use could apply `partitionEithers` to get `([a],[a])` as necessary. Also, if this function is deemed useful, it could be moved to a `CommonUtils.daml` module so it can be used elsewhere.

Reply:
Use `DA.List.groupBy` instead.